src/ensembl/src/content/app/entity-viewer/components/transcript-filter/EntityViewerTranscriptFilter.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import classNames from 'classnames';
+
+import Checkbox from 'src/shared/components/checkbox/Checkbox';
+
+import styles from './EntityViewerTranscriptFilter.scss';
+
+export type Option = {
+  value: string;
+  label: string;
+};
+
+export type OptionsGroup = Option[];
+
+type Props = {
+  selectedValues: string[];
+  onChange: (selectedValues: string[]) => void;
+  options: OptionsGroup[];
+  hideUnchecked?: boolean;
```

Comment:
There seem to be two conflicting tendencies here.

If this is indeed only an `EntityViewerTranscriptFilter`, as the name of the component suggests, then why add the `hideUnchecked` option? Unless I missed something, the prototype doesn't show the case of disappearing unchecked checkboxes.

If, on the other hand, it is a more generic component, where unchecked checkboxes can disappear, then why call it EntityViewerTranscriptFilter?

Incidentally, I think if the api of this component were different, such as something like this:

```
<OptionsFilter className="blah">
  <OptionsGroup className="blah-blah" >
    <Option className="bleh" isSelected={boolean} onChange={(something) => void}>label</Option>
    <Option isSelected={boolean} onChange={(something) => void}>label</Option>
    ....
  </OptionsGroup>
  <OptionsGroup>
    <Option isSelected={boolean} onChange={(something) => void}>label</Option>
    <Option isSelected={boolean} onChange={(something) => void}>label</Option>
    ....
  </OptionsGroup>
</OptionsFilter>
```
then it would be parent's concern whether to show or hide selected options; and the component would also probably be very reusable. I don't know whether we want a reusable filters component — I haven't yet seen them recur in Andrea's designs, but they are almost guaranteed to reappear. Still, a question to Andrea...

Reply:
> why add the hideUnchecked option? Unless I missed something, the prototype doesn't show the case of disappearing unchecked checkboxes.

Closing the filter after making some selections will only show the selected options.

I initially started creating it as a shared component but I had a feeling that a generic shared component like this should probably work more like a Select with popup and a controller.

Particularly in this case, the controller is located on the table header (or a different element) and not attached to the filter itself which is why I though of making this EV specific.

Reply:
> Closing the filter after making some selections will only show the selected options.

Oh, sorry, I didn't realise that was the behaviour. I couldn't find it anywhere in the description, but can now see it in the prototype.

Does this suggest perhaps that what the prop should actually be saying is whether the component is in an "expanded" or "collapsed" state?

Reply:
Using "expanded" or "collapsed" wouldn't make it clear that the collapsed state is going to show all the selected options.

Reply:
The collapsed variant of the component will show what the collapsed variant is expected to show.
The expanded variant of the component will show what the expanded. variant is expected to show.
This is the whole point of declarative vs imperative approach.

For example, the collapsed variant of the filters component is also not showing borders between filter groups. But you did not explicitly instruct the component to do that. It just knows that it shouldn't.

Reply:
To me collapsed would mean more like it is meant to hide everything that is inside, which is not something that happens here.  I used `hideUnchecked` because that's what it does. 
Also, when all the options are selected, there will be nothing to expand/collapse?

Reply:
The intent to expand or collapse the filters is expressed in this arrowhead button though:

![image](https://user-images.githubusercontent.com/6834224/75052367-66508700-54c7-11ea-93aa-fe19ffe033f4.png)

when you click on it, you are not saying to the filters component "please show unchecked filters"; what you are saying to it is "please open".

And when you click on it again:

![image](https://user-images.githubusercontent.com/6834224/75052332-546ee400-54c7-11ea-8efb-69c72d84771a.png)

you effectively say "please close".

I would say the fact that in the closed state the component shows the selected filters is just how it behaves in its closed state 🤷‍♂ 

By the way, this raises the question of whether the component should be "openable" when you have checked all checkboxes. It probably should (in which case it will draw the lines between option groups), but it is still rather confusing 😬